### PR TITLE
Add basic smoke test CI with Travis

### LINKED ISSUE #21: Add tests to validate a new image release
We lack tests to ensure that the modifications we do to the docker files don't break the images, we need to automate or at least document a simple validation test for this. A simple and valuable case is a simple flink job with checkpointing.

### PR DESCRIPTION
**Pull request by patricklucas**
I know #21 suggests running a simple job with checkpointing, but I figure this is a good starting point and we can open a new issue to improve this testing with a basic job with checkpointing.

Closes #21

### CODE REVIEW

**Comment by iemejia** [comment]
I was thinking in how we can expand this in the future to maybe add a minor validation test of a simple wordcount like pipeline, do you have anything in mind for this?

File: testing/smoke_test_all_images.sh [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+CURL_TIMEOUT=1
+CURL_COOLDOWN=1
+CURL_MAX_TRIES=30
+CURL_ENDPOINT=localhost:8081/overview
+NETWORK_NAME=docker-flink-test
+
+function image_tag() {
+    local dockerfile="$1"
+
+    local variant="$(basename "$(dirname "$dockerfile")")"
```
[the remark this answers is not in the document]
Comment (patricklucas) [reply testing/smoke_test_all_images.sh]: Actually I ran into just that in `build_images()`! (hence the separate `local` lines)

We're not using any of the return values in these cases, so I'd just assume use the more concise format for now. It was a good lesson, though. :)

**Comment by patricklucas** [comment]
Yeah I think if you have write access to the repo you'll be able to manage things in Travis. You might want to verify that empirically, though...

I haven't thought about exactly what to test at this point; I don't want to get into the business of testing _Flink_, just that one is able to actually run a working, checkpointing job would be enough I think.

**Comment by patricklucas** [comment]
If you're good with the last commit I added, feel free to merge.

**Comment by patricklucas** [comment]
(The Travis tests here should start passing when this is merged to master, but I didn't want to make the PR history confusing)

**Comment by iemejia** [comment]
Done, thanks for this, I agree with you we should not end up testing more than the minimum that validates things work, not testing flink. Can you please create/update the PR on official images to reflect this.

**Comment by patricklucas** [comment]
Can do. Thanks for the review.

**Pull request merged**